test/e2e/validation_test.go

```diff
@@ -308,11 +308,8 @@ func (tc *testContext) sshSetup() error {
 // runPowerShellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
 // PowerShell, on the host specified by the provided IP.
 func (tc *testContext) runPowerShellSSHJob(name, command, ip string) (string, error) {
-	// Modify command to work when default shell is the newer Powershell version present on Windows Server 2022.
-	powershellDefaultCommand := command
-	if tc.CloudProvider.GetType() == config.VSpherePlatformType || tc.CloudProvider.GetType() == config.AzurePlatformType {
-		powershellDefaultCommand = strings.ReplaceAll(command, "\\\"", "\"")
-	}
+	// Do not scape the double-quotes when default shell is Powershell
+	powershellDefaultCommand := strings.ReplaceAll(command, "\\\"", "\"")
```

Comment:
As WS2019 will be used in the e2e test suite along with the env var `WINDOWS_SERVER_2019_ENABLED`, this logic should be adjusted to capture that.

Reply:
+1 to adding the `WINDOWS_SERVER_VERSION` env var information to testContext struct so eventually this can look like
```suggestion
	powershellDefaultCommand := command
	if tc.windowsServerVersion == "2022" {
		powershellDefaultCommand = strings.ReplaceAll(command, "\\\"", "\"")
	}
```

Reply:
For further references, the double quotes should be adjusted only when the default shell is Powershell.

| Default shell    | OS version    | Need \\" replacement in command?|
| --------------------|-------------------|--------------------------- |
| Windows CMD      | 2019          | No |
| Powershell       | 2019          | No |
| Windows CMD      | 2022          | No |
| Powershell       | 2022          | Yes |

Thanks @saifshaikh48 for sharing this table.